exercises/linked-list/canonical-data.json

```diff
@@ -0,0 +1,532 @@
+{
+  "exercise": "linked-list",
+  "comments": [
+    "In order to keep the interfae for the exercise close or equal to the     ",
+    "description.md and also satisfying the canonical-schema.json, the only   ",
+    "properties are pop, push, shift, unshift. Some tracks also implement     ",
+    "delete and count, via a Track-Inserted hints.                            ",
+    "                                                                         ",
+    "It is hard to have interesting tests using the property based approach so",
+    "this canonical data uses the following approach:                         ",
+    "- Each test input is an array of { operation, value?, expected? }        ",
+    "- If operation is push, unshift, delete, then value is given             ",
+    "- If operation is pop, shift, count, the expected value can be given     ",
+    "                                                                         ",
+    "Encoding tests and operations using the same field is necessary to have  ",
+    "tests that don't just initialize the list, and then have one operation.  "
+  ],
+  "cases": [
+    {
+      "uuid": "7f7e3987-b954-41b8-8084-99beca08752c",
+      "description": "pop gets element from the list",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 7
+          },
+          {
+            "operation": "pop",
+            "expected": 7
+          }
+        ]
+      },
+      "expected": []
```

Comment:
We don't yet have a fixed rule for this, but I recall two exercises which canonical-data was also in a form where the input contains expected values and the expected value should be ignored:

- [`react`](https://github.com/exercism/problem-specifications/blob/master/exercises/react/canonical-data.json#L66)
- [`circular-buffer`](https://github.com/exercism/problem-specifications/blob/master/exercises/circular-buffer/canonical-data.json#L40)

Both these canonical data use an empty object for the expected value. Granted, it doesn't really matter much, but it _may_ be useful to have some consistency.

Reply:
I'm fine with that!